Source/Urho3D/Core/WorkQueue.cpp

```diff
@@ -20,473 +20,563 @@
 // THE SOFTWARE.
 //
 
-#include "../Precompiled.h"
+#include "Urho3D/Precompiled.h"
 
-#include "../Core/CoreEvents.h"
-#include "../Core/ProcessUtils.h"
-#include "../Core/Profiler.h"
-#include "../Core/Thread.h"
-#include "../Core/Timer.h"
-#include "../Core/WorkQueue.h"
-#include "../IO/Log.h"
+#include "Urho3D/Core/WorkQueue.h"
+
+#include "Urho3D/Core/CoreEvents.h"
+#include "Urho3D/Core/ProcessUtils.h"
+#include "Urho3D/Core/Profiler.h"
+#include "Urho3D/Core/Thread.h"
+#include "Urho3D/Core/Timer.h"
+#include "Urho3D/IO/Log.h"
+
+#ifdef URHO3D_THREADING
+#include <enkiTS/src/TaskScheduler.h>
+#endif
 
 namespace Urho3D
 {
 
-/// Thread index.
-static thread_local unsigned currentThreadIndex = M_MAX_UNSIGNED;
-static unsigned maxThreadIndex = 1;
+namespace
+{
 
-/// Worker thread managed by the work queue.
-class WorkerThread : public Thread, public RefCounted
+static unsigned threadIndexCount = 1;
+
+#ifdef URHO3D_THREADING
+static enki::TaskScheduler* taskScheduler = nullptr;
```

Comment:
We really ought to move to `static Context*` instance... Not in this PR, but clearly we need it.